### PR TITLE
Configure gradle build scan plugin

### PR DESCRIPTION
**Pull request by aalmiray**
This small PR configures the `build-scan` Gradle plugin (https://guides.gradle.org/creating-build-scans/) which yields further insight into the build, allowing the team to make informed decisions on where and when the build may be tweaked to get faster, more performant builds.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -7,6 +7,13 @@
 
 import aQute.lib.io.IO
 
+apply plugin: 'com.gradle.build-scan'
+
+buildScan {
+    termsOfServiceUrl   = 'https://gradle.com/terms-of-service'
+    termsOfServiceAgree = 'yes'
```
Comment (bjhargrave) [inline build.gradle]: This also make a blanket assumption the everyone accepts the terms of service. A developer should really make this decision for them self.

**Comment by bjhargrave** [comment]
I don't think we should accept this change.

File: build.gradle (hunk 1 above)
Comment (aalmiray) [reply build.gradle]: I've added a check that solves this problem. Scans are enabled by default when run on CI, but the terms of service are checked if that's not the case. Developers have the choice to accept or refuse the terms.

**Comment by aalmiray** [comment]
Understood. In which case the objection is to enable scans on CI. I can remove the automatic assignment when run on CI as well as the `--scan` flag found in the Travis settings.

**Pull request closed without merging**